enterprise/signals.py

```diff
@@ -135,6 +136,57 @@ def delete_enterprise_learner_role_assignment(sender, instance, **kwargs):     #
             pass
 
 
+@receiver(post_save, sender=EnterpriseCustomerCatalog)
+def update_enterprise_catalog_data(sender, instance, **kwargs):     # pylint: disable=unused-argument
+    """
+    Send data changes to Enterprise Catalogs to the Enterprise Catalog Service.
+    """
+    catalog_uuid = instance.uuid
+    try:
+        catalog_client = EnterpriseCatalogApiClient()
+        response = catalog_client.get_enterprise_catalog(catalog_uuid)
+    except NotConnectedToOpenEdX as exc:
+        logger.exception('Unable to update Enterprise Catalog {}'.format(str(catalog_uuid)), exc)
+    else:
+        if kwargs['created'] or not response:
+            # catalog with matching uuid does NOT exist in enterprise-catalog
+            # service, so we should create a new catalog
+            catalog_client.create_enterprise_catalog(
+                str(catalog_uuid),
+                str(instance.enterprise_customer.uuid),
+                instance.enterprise_customer.name,
+                instance.title,
+                instance.content_filter,
+                instance.enabled_course_modes,
+                instance.publish_audit_enrollment_urls,
+            )
+        else:
+            # catalog with matching uuid does exist in enterprise-catalog
+            # service, so we should update the existing catalog
+            update_fields = {
+                'enterprise_customer': str(instance.enterprise_customer.uuid),
+                'enterprise_customer_name': instance.enterprise_customer.name,
+                'title': instance.title,
+                'content_filter': instance.content_filter,
+                'enabled_course_modes': instance.enabled_course_modes,
+                'publish_audit_enrollment_urls': instance.publish_audit_enrollment_urls,
+            }
+            catalog_client.update_enterprise_catalog(catalog_uuid, **update_fields)
```

Comment:
Out of curiosity, why don't we cast `catalog_uuid` to a string here, like we're doing in the `create_enterprise_catalog()` call?

Reply:
i'm not sure why exactly the client ended up being inconsistent in this way, but after taking a quick glance at the client it appears to be because for non-POST calls, the uuid is part of the path and thus constructed through slumber vs. being part of the post parameter data where everything needs to be a string.